addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,498 @@
+        xsd_id = f'l10n_es_edi_tbai.{invoice.company_id.l10n_es_tbai_tax_agency}_ticketBaiV1-2.xsd'
+        res = {invoice: self._l10n_es_tbai_verify_xml(inv_xml, xsd_id)}
+        if 'error' in res[invoice]:
+            return res
+
+        # Call the web service and get response
+        res.update(self._l10n_es_tbai_post_to_web_service(invoice, inv_xml))
+        response_xml = res[invoice]['response']
+
+        # TIMEOUT / NO RESPONSE: success is undetermined
+        if response_xml is None:
+            pass
+
+        # SUCCESS
+        elif res[invoice].get('success'):
+            # Attachment: post to chatter and save as EDI document
+            invoice.with_context(no_new_invoice=True).message_post(
+                body="<pre>TicketBAI: posted XML\n" + res[invoice]['message'] + "</pre>",
+                attachment_ids=invoice.l10n_es_tbai_post_xml.ids)
+            res[invoice]['attachment'] = invoice.l10n_es_tbai_post_xml
+
+        # FAILURE
+        else:
+            invoice._update_l10n_es_tbai_submitted_xml(xml_doc=None, cancel=False)
+
+        return res
+
+    def _cancel_invoice_edi(self, invoice):
+        # OVERRIDE
+        if self.code != 'es_tbai':
+            return super()._post_invoice_edi(invoice)
+
+        # Configuration check
+        error_dict = self._check_configuration_is_complete(invoice)
+        if error_dict is not None:
+            return {inv: error_dict
+                    for inv in invoice}
+
+        # Generate the XML values.
+        cancel_xml = self._l10n_es_tbai_get_invoice_xml(invoice, cancel=True)
+
+        # Optional check using the XSD
+        xsd_id = f'l10n_es_edi_tbai.{invoice.company_id.l10n_es_tbai_tax_agency}_Anula_ticketBaiV1-2.xsd'
+        res = {invoice: self._l10n_es_tbai_verify_xml(cancel_xml, xsd_id)}
+        if 'error' in res[invoice]:
+            return res
+
+        # Call the web service and get response
+        res.update(self._l10n_es_tbai_post_to_web_service(invoice, cancel_xml, cancel=True))
+        response_xml = res[invoice]['response']
+
+        # TIMEOUT / NO RESPONSE: success is undetermined
+        if response_xml is None:
+            pass
+
+        # SUCCESS
+        elif res[invoice].get('success'):
+            # Put attachment in chatter
+            invoice.with_context(no_new_invoice=True).message_post(
+                body="<pre>TicketBAI: posted cancellation XML\n" + res[invoice]['message'] + "</pre>",
+                attachment_ids=invoice.l10n_es_tbai_cancel_xml.ids)
+
+        # FAILURE
+        else:
+            invoice._update_l10n_es_tbai_submitted_xml(xml_doc=None, cancel=True)  # will need to be re-created
+
+        return res
+
+    # -------------------------------------------------------------------------
+    # TBAI XML VERIFY
+    # -------------------------------------------------------------------------
+
+    def _l10n_es_tbai_verify_xml(self, xml, xsd_id):
+        xsd_attachment = self.env.ref(xsd_id, False)
```

Comment:
always specify the kwargs plz ;) here: `xsd_attachment = self.env.ref(xsd_id, raise_if_not_found=False)`